Source/WebCore/editing/TextIterator.cpp

```diff
@@ -478,7 +478,7 @@ void TextIterator::advance()
             if (!isRendererVisible(renderer.get(), m_behaviors)) {
                 m_handledNode = true;
                 m_handledChildren = !hasDisplayContents(*protectedCurrentNode()) && !renderer;
-            } else if (is<Element>(m_currentNode.get()) && renderer->isSkippedContentRoot()) {
+            } else if (is<Element>(m_currentNode.get()) && (renderer->isSkippedContentRoot() || renderer->style().effectiveInert())) {
```

Comment:
Are we sure this won't affect [`element.innerText`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText)?

Reply:
I hadn’t considered what would happen with [element.innerText](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText) but I’ll test it and see.

By the way: given that per-spec, nodes (not just elements) can be inert, I wonder if we need to have this `renderer->style().effectiveInert()))` check performed on its own on as a separate condition — rather than in combination with checking whether `m_currentNode` is an element node.

And I guess we ideally should have separate test for the inert-node-that’s-not-an-element-node case, if possible. 

It’s not actually immediately clear to me how a node that’s not a element _could_ in fact also be inert — I guess I need to read up on the spec and see (and so I can write up a test for it).

Reply:
Or actually (just thinking out loud) maybe it’s the case that nodes other than element nodes are anyway not going to be exposed to find-in-page (and so wouldn’t matter for the purposes of this patch). Anyway, I’ll read through the spec for inert right now

Reply:
OK I see now that per-spec at https://html.spec.whatwg.org/multipage/interaction.html#modal-dialogs-and-inert-subtrees it’s the case that text nodes can become inert due to being outside an open modal dialog. So we should probably get a test added for that case. Actually, it wouldn’t just be for text nodes outside an open modal dialog, but for element nodes too — because those elements wouldn’t have an explicit `inert` attribute.

I imagine the existing `dialog` implementation already handles making those nodes inert — `effectiveInert()` style — but I’ll look. And I’ll also look at the existing `dialog` tests to see what might be repurposable to use as a starting point for a insert-with-dialog-and-find-in-page test